### ISSUE TITLE
Only use query, not writeToStore。My errors ---- Missing field __typename and other field。

### ISSUE THREAD

**Issue by hytStart** [apollographql/apollo-client#5202]
When i query data from Java backend, they give me some datas which are short of some fields。

Java RD says:  Java Frame missing field when this filed returns null .

![image](https://user-images.githubusercontent.com/34086399/63257574-a5014c80-c2ac-11e9-95a8-d4a5d71fb98d.png)

So that Apollo client give me `data: null`.

Could you advise me on this issue?

```
// client 
export const client = new ApolloClient({
    link: from([errorLink, httpLink]),
    cache,
    defaultOptions: {
        query: {
            fetchPolicy: 'network-only',
            errorPolicy: 'all',
        },
    },
})
// query
const query = gql`
            query myQuery($subOrderSerialNos: String){
                queryOrderDetailInfoBySubOrderSerialNo(subOrderSerialNos: $subOrderSerialNos) {
                    createTime
                    bizDate
                    channelName
                    couponAmount
                    paidAmount
                    orderDetail {
                        itemTitle
                        subOrderSerialNo
                        subOrderStatusDesc
                        itemPrice
                        packageInfo {
                            firstProjectName
                            secondProjectName
                        }
                        legionInfo {
                            consultant
                        }
                        protocolInfo {
                            serviceProtocolUrl
                            rescissionProtocolUrl
                            serviceBeginDate
                            serviceEndDate
                        }
                        payInfo {
                            payDetails {
                                methodCode
                                methodName
                            }
                        }
                        insuranceInfoList {
                            policyNo
                            insuranceName
                            statusDesc
                            insurancePeriod
                            claimPeriod
                            noticeUrl
                            consignUrl
                            premium
                            learningInfo {
                                learnCount
                                scheduleCount
                                learnRate
                            }
                            claimItemList {
                                claimItemName
                                applyBeginTime
                                applyEndTime
                                reportBeginTime
                                reportEndTime
                            }
                        }
                    }
                }
            }
        `
        const params = {
            subOrderSerialNos: '20190626000043_1',
        }
        client.query({
            query,
            variables: params,
        }).then(data => {
            console.log(data)
            dispatch({
                type: 'ORDER_DETAIL_GRAPHQL_SUCCEEDED',
                payload: data,
            })
        }).catch(e => {
            console.log(e)
        })
```

**Comment by kinow** [comment]
I think I have the [same issue](https://github.com/cylc/cylc-ui/issues/420). We are building an API with deltas for changes in workflows states. But the Python API returns a subset of the selection.

It means our query has something like:

```graphql
deltas {
  workflows {
    tasks {}
    jobs {}
  }
}
```

But a response may have:

```graphql
deltas {
  workflows {
    tasks { ... }
    # sorry, no jobs!
  }
}
```

When that happens, I get a similar warning in the browser console.

![image](https://user-images.githubusercontent.com/304786/75723408-21d3af80-5d41-11ea-997e-87f2d749b66f.png)

We might get a lot of these messages, cluttering the browser console. Making it harder to find errors, and apparently causing some slowness when the console is displayed.

From what I could tell by debugging and reading parts of the code, the `console.log` is emitted when writing to the in-memory cache.

https://github.com/apollographql/apollo-client/blob/f26b98cc8ad859bef4ec74d77fd4245b022cc60d/src/cache/inmemory/writeToStore.ts#L240

It will check the `selectionSet`, which will contain `jobs`, and `tasks` (and `__typename`). But it will return `undefined` for jobs, and cause the invariant log. There's no flag in the invariant, so I don't believe it can be disabled in runtime?

Also tried setting the `errorPolicy` or pass the `error` or `onError` to watchquery/subscription functions with no success. I'll try next to find something like a NoOpCache, or a way to completely disable caching and see if the issue goes away. But would be nice if I could simply disable the warning.

**Comment by kinow** [comment]
p.s.: quick hack to be able to keep developing without so many entries in the console output:

```js
console.log = function () {}
console.warn = function () {}
```

**Comment by dwsutherland** [comment]
Would be nice if there's a way to ignore GraphQL violations in the result-data/response, i.e. by setting result validation to `false`.. (Is this possible?)

**Comment by yohan-atlan** [comment]
It's a little bit dirty but try ```fetchPolicy: "no-cache",```

**Comment by kinow** [comment]
> 
> 
> It's a little bit dirty but try `fetchPolicy: "no-cache",`

Hi @yohan-atlan we've always used this option as our system shows workflows statuses, which are hard to cache.

https://github.com/kinow/cylc-ui/blob/f44be4bbe041d8373ccdfc832d750a30321402c9/src/services/workflow.service.js#L65

https://github.com/kinow/cylc-ui/blob/f44be4bbe041d8373ccdfc832d750a30321402c9/src/utils/graphql.js#L114-L123

(links from the branch I am working at the moment to add deltas to a query, and that is using the [hack](https://github.com/kinow/cylc-ui/blob/f44be4bbe041d8373ccdfc832d750a30321402c9/src/services/workflow.service.js#L68) mentioned above to silence the warnings)

From what I recall from quickly looking at the sources in WebStorm, the message would be always logged, even without a cache. It appeared to me that it would still access the InMemoryCache, no matter what settings I had. I tried creating a client with no cache, but it failed. I guess the cache is used internally for query handling perhaps?

I just tested with 2.6.8 - I was using 2.6.4 on that branch - and the issue still happens.
